Tribler/Core/Modules/tracker_manager.py

```diff
@@ -95,6 +95,11 @@ def update_tracker_info(self, tracker_url, is_successful):
         :param tracker_url: The given tracker_url.
         :param is_successful: If the check was successful.
         """
+        # It is possible that a Deferred call could not be cancelled and invokes this function while
+        # we are shutting down, in that case we ignore it.
+        if not self._tracker_dict:
```

Comment:
If it's canceled it won't fire. If it fires it's too late to cancel it and trying to do so would raise an exception, so the task manager checks before trying. In any case I don't see a Deferred here. What call stack triggered the error? This looks like it should be solved elsewhere to me. 
Quick glance shows me that this function is called from two places. And both of them are running on the reactor's thread. so the decorator is also redundant here.